### PR TITLE
GUI-wide settings class

### PR DESCRIPTION
**Pull request by retiutut**
Add GUI-wide settings class to keep certain settings across sessions and app starts #997

Uses pattern established by ADS1299SettingsController.pde which uses GSON and a specific class for storing data values.

### CODE REVIEW

File: OpenBCI_GUI/GuiSettings.pde [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            return true;
+        } catch (IOException e) {
+            e.printStackTrace();
+            outputWarn("OpenBCI_GUI::Settings: Error saving GUI-wide settings to file. Please make an issue on GitHub.");
+            return false;
+        }
+    }
+
+    //Call this method at the end of GUI main Setup in OpenBCI_GUI.pde to make sure everything exists
+    public void applySettingsToFrontEnd() {
```
Comment (Andrey1994) [inline OpenBCI_GUI/GuiSettings.pde:102]: it seems like such methods should be in another class(in classes for particular setting if you already created them), maybe the interface for settings can have a method like `applySettingsToFrontEnd` and these methods should be called from the *main* applySettings method

**Review by Andrey1994 — APPROVED** [review]

File: OpenBCI_GUI/GuiSettings.pde (hunk 1 above)
Comment (retiutut) [reply OpenBCI_GUI/GuiSettings.pde:104]: Ok giving this a try. Pushing a new commit to address this.

Comment (retiutut) [reply OpenBCI_GUI/GuiSettings.pde:104]: I'm getting lots of null pointers trying to move these methods and still maintain functionality.

File: OpenBCI_GUI/GuiSettings.pde [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        topNav.configSelector.toggleExpertModeFrontEnd(values.expertMode.getBooleanValue());
+    }
+
+    public void setExpertMode(ExpertModeEnum val) {
+        values.expertMode = val;
+        saveToFile();
```
Comment (retiutut) [inline OpenBCI_GUI/GuiSettings.pde:110]: Changing a setting will also need to update the settings file. Putting this method in the GuiSettingsValues didn't seem to work.

**Pull request merged**